src/Workspaces/Core/Portable/Editing/ImportAdderService.cs

```diff
@@ -18,44 +20,240 @@ namespace Microsoft.CodeAnalysis.Editing
+                {
+                    continue;
+                }
+
+                var annotations = annotatedNode.GetAnnotations(SymbolAnnotation.Kind);
+                foreach (var annotation in annotations)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    foreach (var namedType in SymbolAnnotation.GetSymbols(annotation, model.Compilation).OfType<INamedTypeSymbol>())
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+                        if (IsBuiltIn(namedType))
+                        {
+                            continue;
+                        }
+
+                        var namespaceSymbol = namedType.ContainingNamespace;
+                        if (namespaceSymbol?.IsGlobalNamespace != false)
+                        {
+                            continue;
+                        }
+
+                        first ??= annotatedNode;
+                        last = annotatedNode;
+
+                        if (addedSymbols.Contains(namespaceSymbol))
+                        {
+                            continue;
+                        }
+
+                        var namespaceSyntax = GenerateNamespaceImportDeclaration(namespaceSymbol, generator);
+
+                        if (addImportsService.HasExistingImport(model.Compilation, root, annotatedNode, namespaceSyntax))
+                        {
+                            continue;
+                        }
+
+                        if(IsInsideNamespace(annotatedNode, namespaceSymbol, model))
+                        {
+                            continue;
+                        }
+
+                        addedSymbols.Add(namespaceSymbol);
+                        importsToAdd.Add(namespaceSyntax);
+                    }
+                }
+            }
+
+            return (importsToAdd, first?.GetCommonRoot(last));
+        }
+
+        private bool IsBuiltIn(INamedTypeSymbol type)
+        {
+            switch (type.OriginalDefinition.SpecialType)
+            {
+                case SpecialType.System_Object:
+                case SpecialType.System_Void:
+                case SpecialType.System_Boolean:
+                case SpecialType.System_Char:
+                case SpecialType.System_SByte:
+                case SpecialType.System_Byte:
+                case SpecialType.System_Int16:
+                case SpecialType.System_UInt16:
+                case SpecialType.System_Int32:
+                case SpecialType.System_UInt32:
+                case SpecialType.System_Int64:
+                case SpecialType.System_UInt64:
+                case SpecialType.System_Decimal:
+                case SpecialType.System_Single:
+                case SpecialType.System_Double:
+                case SpecialType.System_String:
+                case SpecialType.System_Nullable_T:
+                    return true;
```

Comment:
can this bre replaced with `type.IsSpecialType()`?

Reply:
As an aside, the name of that function is confusing, since it implies that Type.SpecialType is not SpecialType.None, but actually it could be eg. DateTime